### ISSUE TITLE
Add ability to add public stacks and recipes for users who have manageCodenvy permission

### ISSUE THREAD

**Issue by sleshchenko** [codenvy/codenvy#1002]
Now admin of system can add public stacks and recipes by predefined configuration. But it is not comfortable to edit it and restart tomcat to apply new changes.
It is needed to allow for "admins" (users who have manageCodenvy permission) add public stacks and recipes by REST API.
REST method that will make stack accessible to everyone is POST `/api/permissions/` with body

```
{
  "userId": "*",
  "domainId": "stack  /  recipe"
  "instanceId": "{STACK_OR_RECIPE_ID_HERE}",
  "actions" : ["search"]
}
```

**Comment by gazarenkov** [comment]
So a User with **manageCodenvy** permission may create stack calling this REST method.

The simplest UI would be checkbox or button like "Make it public" on UD -> Stacks (disabled or throws error message if User does not have a proper permissions).

**Comment by JamesDrummond** [comment]
@gazarenkov The UI would not be available in codenvy.io right? I wouldn't want "Make it public" button visible in codenvy.io unless the message linked to/copy from doc section [https://eclipse-che.readme.io/docs/stacks#adding-stacks-to-the-che-default-assembly](https://eclipse-che.readme.io/docs/stacks#adding-stacks-to-the-che-default-assembly).

**Comment by slemeur** [comment]
Will be addressed in #1902 epic
